src/tests/class_tests/openms_gui/source/GUI/TVPrefDialog_test.cpp

```diff
@@ -0,0 +1,352 @@
+  QTest::qWait(DELAY);
+
+  // test path input methods
+  
+  // file dialog
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkFileDialog_);
+  QTest::mouseClick(UI->browse_default, Qt::LeftButton, 0, QPoint());
+  QTest::qWait(DELAY);
+
+  // line edit
+  QTest::keyClicks(UI->default_path, "C:\dev");
+  QVERIFY2(UI->default_path->text() == "C:\dev", "Line edit for default path broken.");
+  QTest::qWait(DELAY);
+
+  // test check boxes
+  testCheckBox(UI->default_path_current);
+  testCheckBox(UI->use_cached_ms1);
+  testCheckBox(UI->use_cached_ms2);
+
+  // TODO: 'map_default' (Qt::ComboBox), 'map_cutoff' (Qt::ComboBox), 'on_file_change' (Qt::ComboBox)
+
+  /////////////////////////////////////////////
+  //             '1D view' tab               //
+  /////////////////////////////////////////////
+
+  // click on tab
+  QTest::mouseClick(tab_bar, Qt::LeftButton, 0, tab_bar->tabRect(ONE_D).center());
+  QTest::qWait(DELAY);
+
+  // test color selectors
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkColorDialog_);
+  QTest::mouseClick(UI->color_1D, Qt::LeftButton, 0, QPoint());
+  QTest::qWait(DELAY);
+
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkColorDialog_);
+  QTest::mouseClick(UI->selected_1D, Qt::LeftButton, 0, QPoint());
+  QTest::qWait(DELAY);
+
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkColorDialog_);
+  QTest::mouseClick(UI->icon_1D, Qt::LeftButton, 0, QPoint());
+  QTest::qWait(DELAY);
+
+  /////////////////////////////////////////////
+  //             '2D view' tab               //
+  /////////////////////////////////////////////
+
+  // click on tab
+  QTest::mouseClick(tab_bar, Qt::LeftButton, 0, tab_bar->tabRect(TWO_D).center());
+  QTest::qWait(DELAY);
+
+  // test spin box
+  UI->feature_icon_size_2D->clear();
+  QTest::keyClicks(UI->feature_icon_size_2D, "5");
+  QTest::qWait(DELAY);
+  QVERIFY(5 == UI->feature_icon_size_2D->value());
+
+  // TODO:: 'peak_2D' (OpenMS::MultiGradientSelector), 'mapping_2D' (Qt::ComboBox), 'feature_icon_2D' (Qt::ComboBox)
+  
+  /////////////////////////////////////////////
+  //             '3D view' tab               //
+  /////////////////////////////////////////////
+
+  // click on tab
+  QTest::mouseClick(tab_bar, Qt::LeftButton, 0, tab_bar->tabRect(THREE_D).center());
+  QTest::qWait(DELAY);
+
+  // test spin box
+  UI->line_width_3D->clear();
+  QTest::keyClicks(UI->line_width_3D, "2");
+  QTest::qWait(DELAY);
+  QVERIFY(2 == UI->line_width_3D->value());
+
+  // TODO: 'shade_3D' (Qt::ComboBox), 'peak_3D' (OpenMS::MultiGradientSelector)
+
+  /////////////////////////////////////////////
+  //               'TSG' tab                 //
+  /////////////////////////////////////////////
+
+  // click on tab
+  QTest::mouseClick(tab_bar, Qt::LeftButton, 0, tab_bar->tabRect(TSG).center());
+  QTest::qWait(DELAY);
+
+  // test spin box
+  UI->tolerance->clear();
+  QTest::keyClicks(UI->tolerance, "0,5");
```

Comment:
`0,5` may not work on all locales?!
I'm surprised it works now honestly, because we should be using a c-locale by default and I would have expected `0.5` to work.

Reply:
`0,5` is the keyboard input for the spin box - it doesn't support typing a `.`
I'm unsure if this differs depending on system language...

Reply:
could you give that some google/bing/duckduckgo time?

Reply:
already on it 👍